src/ort_genai_c.cpp

```diff
@@ -214,6 +214,36 @@ void OGA_API_CALL OgaTokenizerDestroyStrings(const char** strings, size_t count)
   delete strings;
 }
 
+OgaResult* OGA_API_CALL OgaTokenizerEncodeBatchStrings(const OgaTokenizer* p, const OgaStrings* string_array, OgaSequences** out) {
+  OGA_TRY
+  auto& tokenizer = *reinterpret_cast<const Generators::Tokenizer*>(p);
+  auto& strings = *reinterpret_cast<const std::vector<std::string>*>(string_array);
+  *out = reinterpret_cast<OgaSequences*>(std::make_unique<Generators::TokenSequences>(std::move(tokenizer.EncodeBatch(strings))).release());
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaTokenizerDecodeBatchStrings(const OgaTokenizer* p, const OgaSequences* p_sequences, OgaStrings** out_strings) {
+  OGA_TRY
+  auto& tokenizer = *reinterpret_cast<const Generators::Tokenizer*>(p);
+  auto& sequences = *reinterpret_cast<const Generators::TokenSequences*>(p_sequences);
+
+  *out_strings = reinterpret_cast<OgaStrings*>(
+      std::make_unique<std::vector<std::string>>(tokenizer.DecodeBatch(sequences)).release());
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaTokenizerEncode(const OgaTokenizer* p, const char* str, OgaSequences** out) {
+  OGA_TRY
+  auto& tokenizer = *reinterpret_cast<const Generators::Tokenizer*>(p);
+  auto out_tokens = std::make_unique<Generators::TokenSequences>();
+  out_tokens->emplace_back(std::move(tokenizer.Encode(str)));
```

Comment:
Move does not seem to be necessary

Reply:
Move was not necessary here. Removed it.